third_party/helm/helm.go

```diff
@@ -0,0 +1,208 @@
+		opts.Chart = chart
+	}
+}
+
+// WithMode is used to invoke the first sub command of the helm binary. This can
+// be configured to run any helm command while used with WithArgs helper
+func WithMode(mode string) Option {
+	return func(opts *Opts) {
+		opts.Mode = mode
+	}
+}
+
+// WithArgs is used to inject additional arguments into the Helm commands.
+// Please pay careful consideration while using this as the current one
+// does not have the ability to de-dup the arguments since options such as
+// -f or --set can be used multiple times with different values, and we wanted
+// the end user to make the decision of how the commands are invoked and not
+// restrict them to a specific way of invoking the commands.
+func WithArgs(args ...string) Option {
+	return func(opts *Opts) {
+		opts.Args = append(opts.Args, args...)
+	}
+}
+
+// WithWait is used to configure the argument used by the helm to indicate it
+// should wait for an acceptable state of the resource before yielding the
+// control back to the invocation point
+func WithWait() Option {
+	return func(opts *Opts) {
+		opts.Wait = true
+	}
+}
+
+// WithTimeout is used to configure the time for which the helm command should
+// be in blocked wait mode for the acceptable state of the resource to be reached
+func WithTimeout(timeout string) Option {
+	return func(opts *Opts) {
+		opts.Timeout = timeout
+	}
+}
+
+// processOpts is used to generate the Opts resource that will be used to generate
+// the actual helm command to be run using the getCommand helper
+func (m *Manager) processOpts(opts ...Option) *Opts {
+	option := &Opts{}
+	for _, op := range opts {
+		op(option)
+	}
+	return option
+}
+
+// getCommand is used to convert the Opts into a helm suitable command to be run
+func (m *Manager) getCommand(opt *Opts) (string, error) {
+	commandParts := []string{"helm", opt.Mode}
+	if opt.Mode == "" {
+		return "", fmt.Errorf("missing helm operation mode. Please use the WithMode option while invoking the Run")
+	}
+	if opt.Name != "" {
+		commandParts = append(commandParts, opt.Name)
+	}
+	if opt.Chart != "" {
+		commandParts = append(commandParts, opt.Chart)
+	} else {
+		commandParts = append(commandParts, opt.ReleaseName)
+	}
+	if opt.Namespace != "" {
+		commandParts = append(commandParts, "--namespace", opt.Namespace)
+	}
+	if opt.Version != "" {
+		commandParts = append(commandParts, "--version", opt.Version)
+	}
+	commandParts = append(commandParts, opt.Args...)
+	if opt.Wait {
+		commandParts = append(commandParts, "--wait")
+	}
+	if opt.Timeout != "" {
+		commandParts = append(commandParts, "--timeout", opt.Timeout)
+	}
+	commandParts = append(commandParts, "--kubeconfig", m.kubeConfig)
+	return strings.Join(commandParts, " "), nil
+}
+
+// Run method is used to invoke a helm command to perform a suitable operation.
+// Please make sure to configure the right Opts using the Option helpers
+func (m *Manager) Run(opts ...Option) (proc *exec.Proc, err error) {
```

Comment:
Avoid leaking type `*exec.Proc` which would force tests to depend on `gexe`. Instead, handle all errors, using gexe.Proc (i.e. proc.Error, proc.IsSuccess, etc) within the confined of the Run method and only return an error if something goes wrong.

Reply:
The idea behind this was to help user/test workflow where they might want to log the outputs. But I can also take the approach of `log.V(4).Info` and dump it here and return the exit codes alone. Let me take care of it. Thanks